file/builder.go

```diff
@@ -430,14 +430,6 @@ func (b *stateBuilder) services() {
 		utils.MustMergeTags(&s.Service, b.selectTags)
 		b.defaulter.MustSet(&s.Service)
 
-		if s.ClientCertificate != nil && !utils.Empty(s.ClientCertificate.ID) {
-			if _, ok := b.certIDs[*s.ClientCertificate.ID]; !ok {
```

Comment:
It seems `b.certIDs` was used only in this one place and can now be completely removed.
Can you delete the following lines as well?
```
20:     certIDs      map[string]bool
48:     b.certIDs = map[string]bool{}
109:            b.certIDs[*c.ID] = true
```

Reply:
Ah yes, true!  Have removed those lines.